src/lib.rs

```diff
@@ -28,6 +28,7 @@ fn s_err(span: proc_macro2::Span, msg: impl fmt::Display) -> syn::Error {
 /// - `from` to specify a conversion function from repr to the bitfield's integer type
 /// - `into` to specify a conversion function from the bitfield's integer type to repr
 /// - `debug` to disable the `Debug` trait generation
+/// - `defmt` to enable the `defmt::Format` trait generation.
```

Comment:
There are multiple popular formatting crates for embedded, and usually you only use one of them.
Maybe it would be better to only have the `debug` attribute but with different values for the formatting crates:

```rust
#[bitfield(u16, debug = None)]
struct Foo { /*...*/ }
#[bitfield(u16, debug = Core)] // <- Default
struct Foo { /*...*/ }
#[bitfield(u16, debug = Defmt)]
struct Foo { /*...*/ }
#[bitfield(u16, debug = Ufmt)]
struct Foo { /*...*/ }
// ...
```

Reply:
Should `debug = Defmt` still implement `core::fmt::Debug` also? I usually implement both in my crates, simply because `Debug` can be relied upon to almost always exist. If a bitfield ends up buried inside a foreign type I want to look at, I don't really want to have to remove the `Defmt` implementation (and any existing code that uses it) for `Debug` to work.

Reply:
Something I just remembered: a lot of crates (mine included) put `defmt` implementations behind a feature flag. Using something like `defmt = cfg!(feature = "defmt")` doesn't work, because the proc macro expects a boolean literal there, and I suspect the same will happen with the enum argument.

Instead, I do something like this:

~~~~
#[cfg_attr(not(feature = "defmt"), bitfield(u16))]
#[cfg_attr(feature = "defmt", bitfield(u16, defmt = true))]
pub struct Register { .. }
~~~~

This is fine, but awkward, and will only get worse if I had more arguments to `bitfield`. Can you think of a better way? Maybe a `debug-feature` argument?

Reply:
Ok if it's a realistic use case to have multiple different formatting implementations then the enum approach doesn't really work. Alternatives are multiple bool attributes or accepting multiple values for the debug attribute. The first is more backwards compatible, so let's just keep it.
For expressions beyond bool literals we would have to evaluate them first.